### PR TITLE
Extend kick event for automatic fly kicks

### PR DESCRIPTION
**Pull request by mergu**

### CODE REVIEW

File: plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerKickedScriptEvent.java [hunk 1]
```diff
@@ -63,28 +75,38 @@ public String getName() {
...
-        }
-        else if (name.equals("reason")) {
-            return reason;
+        switch (name) {
+            case "message":
+                return message;
+            case "reason":
+                return reason;
+            case "flying":
+                return new Element(isFlying());
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerKickedScriptEvent.java:109]: Don't reformat code that appears in 100 places without justifying it and applying it to all relevant cases.

Comment (mergu) [reply plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerKickedScriptEvent.java:109]: This was just an automatic suggestion by IntelliJ. I don't see any harm in changing these as we go along

**Comment by mergu** [comment]
This is tested & ready for merge btw

**Comment by mcmonkey4eva** [comment]
This is not ready for merge, refer to unresolved comments.

File: plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerKickedScriptEvent.java [hunk 2]
```diff
@@ -40,17 +44,25 @@ public PlayerKickedScriptEvent() {
...
+    }
+
     @Override
     public boolean couldMatch(ScriptContainer scriptContainer, String s) {
         return CoreUtilities.toLowerCase(s).startsWith("player kicked");
     }
 
     @Override
-    public boolean matches(ScriptPath path) {
+    public boolean matches(ScriptContainer scriptContainer, String s) {
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/net/aufdemrand/denizen/events/player/PlayerKickedScriptEvent.java:62]: Why was this change made?

**Comment by mcmonkey4eva** [comment]
manually merged with a few corrections applied -> https://github.com/DenizenScript/Denizen-For-Bukkit/commit/f1a74df29ef268bf8d19babe1d650afb127f5ec9

**Pull request closed without merging**